packages/xdl/src/ProjectAssets.ts

```diff
@@ -334,18 +336,20 @@ async function collectAssets(
   // Resolve manifest assets to their hosted URL and add them to the list of assets to
   // be uploaded. Modifies exp.
   const manifestAssets: Asset[] = [];
-  await resolveManifestAssets({
-    projectRoot,
-    manifest: exp,
-    async resolver(assetPath) {
-      const absolutePath = path.resolve(projectRoot, assetPath);
-      const contents = await fs.readFile(absolutePath);
-      const hash = md5hex(contents);
-      manifestAssets.push({ files: [absolutePath], fileHashes: [hash], hash });
-      return urljoin(hostedAssetPrefix, hash);
-    },
-    strict: true,
-  });
+  if (exp.sdkVersion) {
+    await resolveManifestAssets({
```

Comment:
If you dig through `resolverManifestAssets` it ends with a call to https://github.com/expo/expo-cli/pull/3547/files#diff-96a934c1e14024d198b87db4ce862a393b9829d562f3d7f0e1c2048eb4153a62R63

which requires an `sdkVersion`